core/running/test-runner.ts

```diff
@@ -44,10 +45,19 @@ export class TestRunner {
        await this._runTests(testSetRunInfo, testSetResults);
     }
 
-    private async _runTests(testSetRunInfo: TestSetRunInfo, results: TestSetResults) {
+    /**
+     * Defined the call back function to be called when the test is completed
+     */
+    public onTestComplete( testCompleteCB: IOnTestCompleteCBFunction) {
```

Comment:
I think this should probably push to an array as if this is called multiple times only the most recent callback will be given the event.